src/Sparrow/Platform/Posix/PosixMemoryQueryMethods.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Sparrow.Exceptions;
+
+namespace Sparrow.Platform.Posix
+{
+    public static unsafe class PosixMemoryQueryMethods
+    {
+        public static bool IsMemoryResident(IntPtr addr, long length)
+        {
+            Debug.Assert(addr.ToInt64() % Syscall.PageSize == 0);
+
+            var isResidentInRam = true;
+            var vecSize = (length + Syscall.PageSize - 1) / Syscall.PageSize;
+            var vec = Marshal.AllocHGlobal((int)vecSize);
```

Comment:
Better to define a `stackalloc int[4]` for the common (small) usage, and only allocate on the heap if we really have to.